### PR TITLE
[docs] Update /analyze and wrong credentials in REST API section

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Add /analyze API docs in table and column section
- Update wrong credentials section with 401 unauthorised

### CODE REVIEW

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 1]
```diff
@@ -248,6 +248,12 @@ Describe table API:
 
 - **source_type:** select the configured databases (e.g. `hive`) or connector ID (e.g. `1`)
 
+Analyze API:
+
+    curl -X POST 'https://demo.gethue.com/api/<DIALECT>/analyze/<DB>/<TABLE>/' -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjI1ODE1MzU5LCJqdGkiOiIzODM0M2Q3YjRjNGY0NTUxYWJmYmQyZmI4YjUzYjFjZiIsInVzZXJfaWQiOjExMDA3MTR9.fl0h7VooLtWnu9v7FtdLUy3NukwFtUya-LkTzollTig' -H 'Content-Type: application/x-www-form-urlencoded' -d ''
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:253]: `-d ''` needed? (we could just have a default value if really needed)

File: docs/docs-site/content/developer/api/rest/_index.md [hunk 2]
```diff
@@ -111,13 +111,13 @@ The API authenticates via the [authentication backends](/administrator/configura
 
 Then a JWT token is returned and needs to be passed as a bearer in the headers for all the API calls.
 
-**Wrong credentials**: there is currently no error on bad authentication but instead a 302 redirect to the login page, e.g.:
+**Wrong credentials**: on bad authentication, it will return a `401 unauthorized` response, e.g.:
```
Comment (romainr) [inline docs/docs-site/content/developer/api/rest/_index.md:114]: Nice, indeed it is cleaner now with the new API

**Review by romainr — APPROVED** [review]

File: docs/docs-site/content/developer/api/rest/_index.md (hunk 1 above)
Comment (Harshg999) [reply docs/docs-site/content/developer/api/rest/_index.md]: Removed

File: docs/docs-site/content/developer/api/rest/_index.md (hunk 2 above)
Comment (Harshg999) [reply docs/docs-site/content/developer/api/rest/_index.md:114]: Agreed!

**Pull request merged**